source/common/common/mem_block_builder.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
```

Comment:
Unrelated, but I'd also like to see the `memcpy(a, &b, sizeof(b))` be replaced by a macro or template. These are more frequent than the dynamic append case that you handle below, and are generally safe, but it would be nice to make them disappear from memcpy grep and sometimes there are bugs, e.g. sizeof(*p) vs. sizeof(p).

Reply:
https://github.com/envoyproxy/envoy/issues/9328